modules/tinymce/src/core/test/ts/browser/UiEnabledModeTest.ts

```diff
@@ -0,0 +1,1431 @@
+import { Keys, TestStore, UiFinder } from '@ephox/agar';
+import { beforeEach, context, describe, it } from '@ephox/bedrock-client';
+import { Arr, Fun } from '@ephox/katamari';
+import { Attribute, SugarBody } from '@ephox/sugar';
+import { TinyHooks, TinyUiActions } from '@ephox/wrap-mcagar';
+import { assert } from 'chai';
+
+import Editor from 'tinymce/core/api/Editor';
+import { ToolbarMode } from 'tinymce/core/api/OptionTypes';
+import { Dialog } from 'tinymce/core/api/ui/Ui';
+
+describe('browser.tinymce.core.UiEnabledModesTest', () => {
+  const registerMode = (ed: Editor) => {
+    ed.mode.register('testmode', {
+      activate: Fun.noop,
+      deactivate: Fun.noop,
+      editorReadOnly: {
+        uiEnabled: true,
+        selectionEnabled: true,
+      },
+    });
+  };
+
+  const assertMenuEnabled = (menu: string) => {
+    const menuButton = UiFinder.findIn(SugarBody.body(), `.tox-mbtn:contains("${menu}")`).getOrDie();
+    assert.equal(Attribute.get(menuButton, 'disabled'), undefined, 'Should not be disabled');
+  };
+
+  const assertMenuDisabled = (menu: string) => {
+    const menuButton = UiFinder.findIn(SugarBody.body(), `.tox-mbtn:contains("${menu}")`).getOrDie();
+    assert.equal(Attribute.get(menuButton, 'disabled'), 'disabled', 'Should be disabled');
+  };
+
+  const assertButtonEnabled = (selector: string) => UiFinder.notExists(SugarBody.body(), `[data-mce-name="${selector}"][aria-disabled="true"]`);
+
+  const assertButtonDisabled = (selector: string) => UiFinder.exists(SugarBody.body(), `[data-mce-name="${selector}"][aria-disabled="true"]`);
+
+  const assertOverflowButtonDisabled = () => UiFinder.exists(SugarBody.body(), '[data-mce-name="overflow-button"][disabled="disabled"]');
+
+  const assertButtonNativelyDisabled = (selector: string) => UiFinder.exists(SugarBody.body(), `[data-mce-name="${selector}"][disabled="disabled"]`);
+
+  const assertButtonNativelyEnabled = (selector: string) => UiFinder.exists(SugarBody.body(), `[data-mce-name="${selector}"]:not([disabled="disabled"])`);
+
+  const setupButtonsScenario = [
+    {
+      label: 'Normal toolbar button',
+      buttonSetupOne: (ed: Editor) => {
+        ed.ui.registry.addButton('t1', {
+          icon: 'italic',
+          text: 'Test Menu Item 1',
+          onAction: Fun.noop
+        });
+      },
+      buttonSetupTwo: (ed: Editor) => {
+        ed.ui.registry.addButton('t2', {
+          icon: 'italic',
+          text: 'Test Menu Item 2',
+          onAction: Fun.noop,
+          onSetup: (api) => {
+            api.setEnabled(true);
+            return Fun.noop;
```

Comment:
Isn't this the default? You should be able to remove these `return` statements from the test.